estuary/authorization.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from __future__ import unicode_literals
+
+import ssl
+
+from flask import current_app
+from werkzeug.exceptions import InternalServerError
+
+from estuary import log
+
+
+def is_user_authorized(username, employee_type):
+    """
+    Verify the user is authorized to access the application.
+
+    :param str username: the username from the user's token
+    :param str employee_type: the employee type from the user's token
+    :return: a boolean that determines if the user is authorized
+    :rtype: bool
+    """
+    employee_types = current_app.config.get('EMPLOYEE_TYPES', [])
+    if employee_type in employee_types:
+        log.debug('The user %s is an employee', username)
+        return True
+
+    ldap_group_dn = current_app.config.get('LDAP_EXCEPTIONS_GROUP_DN')
+    if ldap_group_dn and username in _get_exception_users():
+        log.debug('The user %s is not considered an employee but is an exception', username)
+        return True
+
+    return False
+
+
+def _get_exception_users():
+    """
+    Get the list of users that are explicitly whitelisted.
+
+    If the LDAP search fails, an empty set is returned.
+
+    :return: a set of usernames
+    :rtype: set
+    :raise InternalServerError: if a required configuration value is not set or the connection to
+        the LDAP server fails
+    """
+    # Import this here so it's not required for deployments with auth disabled
+    import ldap3
```

Comment:
Why can't this be put at the top of the file regardless of whether auth is disabled or not? If auth was disabled, wouldn't Estuary not use this file anyway?

Reply:
This function is used by `estuary.utils.general` regardless of if authentication is enabled or not. I'll import it in `login_required` instead and move this to the top.

Reply:
Actually, this way it allows you to enable authentication and not require `ldap3` to be installed if the configuration `LDAP_EXCEPTIONS_GROUP_DN` is not set. I'll leave it as is.